src/Framework/TestSuite.php

```diff
@@ -183,10 +183,15 @@ public function __construct($theClass = '', $name = '')
         }
 
         foreach ($theClass->getMethods() as $method) {
+            // how can we handle this better? In our code base we have a base class we extend with lots of convenience methods that would not be skipped here and eventually get passed to addTestMethod
```

Comment:
Maybe with `is_a`?

Reply:
We could check whether the class that declares the method is abstract.

Reply:
Strike that, that does not work.

Reply:
@sebastianbergmann what happened when you continued on classes that are abstract? In my case that would work. We have a base test class that looks like this:

```
<?php

use PHPUnit\Framework\TestCase;

abstract class OpenSkyTestCase extends TestCase
{
    // ...
}

Reply:
But now that I am thinking about it, it would break other real scenarios for people where they have a base abstract class with test methods that run as a part of a child class. We do this in our code base too.

Reply:
Exactly :-)